api/src/Mailbox.ts

```diff
@@ -26,8 +26,25 @@ export class GearMailbox {
    * console.log(mailbox.toHuman());
    * ```
    */
-  async read(accountId: Hex | AccountId32 | string): Promise<Option<BTreeMap<H256, QueuedMessage>>> {
-    return this.api.query.gear.mailbox(accountId);
+  async read(accountId: Hex | AccountId32 | string, messageId?: Hex | H256): Promise<IMailbox> {
+    if (messageId) {
+      const mailbox = await this.api.query.gear['mailbox'](accountId, messageId);
+      return mailbox.toHuman() as any;
+    } else {
+      const keys = await this.api.query.gear['mailbox'].keys(accountId);
+      if (keys.length === 0) {
+        return [];
+      }
+      const keyPrefixes = this.api.query.gear['mailbox'].keyPrefix(accountId);
+      const keysPaged = await this.api.rpc.state.getKeysPaged(keyPrefixes, 1000, keyPrefixes);
+      const mailbox = (await this.api.rpc.state.queryStorageAt(keysPaged)) as Option<StoredMessage>[];
+      return mailbox.map((value, index) => {
+        return [
+          keys[index].toHuman() as [AccountId, Hex],
+          this.api.createType('GearCoreMessageStoredStoredMessage', value.unwrap()).toHuman() as any,
```

Comment:
```suggestion
      const mailbox = await this.api.query.gear.mailbox(accountId, messageId);
      return mailbox.toHuman() as IMailbox;
    } else {
      const keys = await this.api.query.gear.mailbox.keys(accountId);
      if (keys.length === 0) {
        return [];
      }
      const keyPrefixes = this.api.query.gear.mailbox.keyPrefix(accountId);
      const keysPaged = await this.api.rpc.state.getKeysPaged(keyPrefixes, 1000, keyPrefixes);
      const mailbox = (await this.api.rpc.state.queryStorageAt(keysPaged)) as Option<StoredMessage>[];
      return mailbox.map((option, index) => {
        return [
          keys[index].toHuman() as [AccountId, Hex],
          this.api.createType('GearCoreMessageStoredStoredMessage', option.unwrap()).toHuman() as unknown as HumanedMessage,
```
Still seems hacky as hell, but it's better than `any`